pkg/operator/operator.go

```diff
@@ -179,7 +179,7 @@ func (o *Operator) Start(ctx context.Context) {
 
 func knativeChecker(path string) healthz.Checker {
 	return func(req *http.Request) (err error) {
-		res, err := http.Get(fmt.Sprintf("http://:%d/%s", knativeinjection.HealthCheckDefaultPort, path))
+		res, err := http.Get(fmt.Sprintf("http://localhost:%d/%s", knativeinjection.HealthCheckDefaultPort, path))
```

Comment:
I can't seem to find an RFC that defines what an empty hostname (or host) value is meant to signify. Indeed this is allowed via `net/http` (via `net/url`):

- https://go.dev/play/p/umQUZuCXbXc

Wondering what use-cases are dependent upon golang allowing an empty host.

In case it's not clear this change lgtm.